### DISCUSSION TITLE
Lets consider integration with Collabora Online

### DISCUSSION THREAD

**Discussion by belonesox** [sagemathinc/cocalc#7021] [Ideas]
Lets consider integration with Collabora Online
https://github.com/CollaboraOnline/online

On first sight, it looks technically possible — like integration with Code-Server (yet another «named project server», proxying port, etc).
It use CollaboraOnline in my nextcloud setups and it looks good for slides-docs-spreadsheets (yes, cocalc has own slides, but may be better not reinvent a wheel with spreadsheets).

Did anyone try it / think about? May be some reasons exist here  why it is not possible?

**Comment by williamstein** [comment]
> Did anyone try it / think about? 

I'm not aware of anybody thinking about it.  Nobody has ever asked us for it in support or in this issue tracker before, as far as I know.  I'm not even sure I've ever heard of Collabora Online before.

> On first sight, it looks technically possible — like integration with Code-Server (yet another «named project server», proxying port, etc).

Certainly doing something like this is technically possible as long as they support a base path, and I don't see any good reason not to.  That said, do they have a model that requires signing in as a user?  If so, how would that map to a cocalc project and cocalc user identity?   None of the existing named servers (jupyterlab, jupyter classic, code server, pluto) have any notion of users at all as a requirement to use them, which makes integration particularly simple.

**Reply by belonesox** [reply]
OK, I will try to conduct research on it.  (I wish to make cocalc as ultimate knowledge management /collaboration tool).

Collabora should works ok as «project user», allowing  collaborative editing 
![image](https://github.com/sagemathinc/cocalc/assets/1609739/2172be57-b8d7-4907-ae6e-4c626d0928e7)

(yes, loosing information what cocalc user actully did changes, but it OK for first try).
